### PR TITLE
Discourage new Case files

### LINKED ISSUE #1089: Cases files are super rigorous to work with
It's so hard to read the unit tests. I'm opening one and need to tangle my way down on so many files without proper IDE support.

```kotlin
val path = Case.UnnecessaryAbstractClassPositive.path()
val findings = subject.lint(path)
```

I can click on `UnnecessaryAbstractClassPositive` which is fine. Then I'm seeing okay I need to go there `/cases/UnnecessaryAbstractClassPositive.kt`. This I can't click with the IDE and need to manually go down.

To make things worse there are two `UnnecessaryAbstractClassPositive.kt` classes.

```
./detekt-rules/src/test/resources/cases/UnnecessaryAbstractClassPositive.kt
./detekt-rules/build/resources/test/cases/UnnecessaryAbstractClassPositive.kt
```

Luckily they are the same though. By the time I'm at the case file I've already lost what I wanted to look up.

Even worse is the fact that a single unit test often checks everything. Some are split up between Positive and Negative although it does not make much of a difference.

Some of the Postive files then help themselves with comments like `// violation: no abstract members`. I don't know whether I can rely on them or not since sometimes they are present sometimes not and I would not expect them to be updated when tweaking a rule. Those things slip understandably in PR reviews.

What I propose instead is to have everything in one file and only one violation being tested at a time. With Android Lint this looks like this:

```kotlin
lint()
    .files(xml("res/layout/ids.xml", """
      <TextView
          xmlns:android="http://schemas.android.com/apk/res/android"
          android:paddingTop="16dp"
          android:paddingBottom="16dp"
          android:paddingStart="16dp"
          android:paddingEnd="16dp"/>""").indented())
    .issues(ISSUE_SUPERFLUOUS_PADDING_DECLARATION)
    .run()
    .expect("""
      |res/layout/ids.xml:1: Warning: Should be using padding instead. [SuperfluousPaddingDeclaration]
      |<TextView
      |^
      |0 errors, 1 warnings""".trimMargin())
```

or 

```kotlin
lint()
    .files(xml("res/layout/ids.xml", """
      <TextView
          xmlns:android="http://schemas.android.com/apk/res/android"
          android:paddingTop="8dp"
          android:paddingBottom="16dp"
          android:paddingStart="16dp"
          android:paddingEnd="16dp"/>""").indented())
    .issues(ISSUE_SUPERFLUOUS_PADDING_DECLARATION)
    .run()
    .expectClean()
```

It's a bit more verbose and we probably don't need a full testing support like there is in Lint. However, it's very clear why and where things are failing and you have everything inside your Junit4 `@Test` method or in your `it` when using Spek.

Bonus points are:

- That we can probably find a way to test out where exactly it's being reported (either by using `^` or `~~~~~~` for longer flags.
- We can also assert our error messages. We had a lot of typos there in the past.
- Tests get decoupled and we can give them better descriptions
- It's easier for everyone to read the tests and adjust it for future changes
- Proper IDE support

### PR DESCRIPTION
**Pull request by 3flex**
Closes #1089 

That issue will remain open until all case files are retired, which may take a long time as it's quite a thankless task - instead, add a comment asking contributors not to add new case files with a reference to the PR so the PR can be closed.

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/Case.kt [hunk 1]
```diff
@@ -4,6 +4,8 @@ import io.gitlab.arturbosch.detekt.test.resource
 import java.nio.file.Path
 import java.nio.file.Paths
 
+/* Do not add new elements to this file.
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/Case.kt]: Lets encourage the usage of inline snippets here?

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/Case.kt]: I agree with Artur.
Furthermore, we should also strive to inline the case files when we change something in a rule or in the corresponding spec file, which still links a case file.

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/Case.kt]: I also try to mention this in PR reviews.

Comment (3flex) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/Case.kt]: I've reworded, hopefully the updated text is OK?

**Pull request merged**